pyalgotrading/algobulls/connection.py

```diff
@@ -81,6 +83,79 @@ def set_access_token(self, access_token):
         assert isinstance(access_token, str), f'Argument "access_token" should be a string'
         self.api.set_access_token(access_token)
 
+    def set_generative_ai_keys(self, api_key, secret_key, token_key):
+        """
+        Set the API keys of the generative AI took used
+
+        Args:
+            api_key: api key name
+            # todo: learn about different generative AIs and how many and what keys are required ?
+                #  also confirm with backend the format of API keys to be received
+        """
+        assert isinstance(api_key, str), f'Argument "api_key" should be a string'
+        self.api.openai_key = api_key
+        return 'SUCCESS' or 'FAILURE'
```

Comment:
need to understand this line

Reply:
Yes, so currently we don't have a way to directly verify if the given key is valid, we won't find out till the first chat is initiated. So need to decide on logic if it's a success or failure. Will remove this later. we are still finding a way to efficiently verify the key.